### PR TITLE
chore: refactor schedule banner

### PR DESCRIPTION
**Pull request by presleyp**
- move more logic from component to XState
- logically, the schedule banner machine is part of the workspace page machine because it depends on `deadline`, which comes from `workspace`. But the workspace page machine is huge. So instead of inlining the schedule banner machine inside the workspace page machine, I `invoke` it. This requires breaking the workspace page component into two components, because you can use a hook (to access the child machine) inside a conditional (which ensures the child machine has been invoked). 
- fix color of Auto-Stop switch to match color of Auto-Start switch and not look disabled

I notice there's a pre-existing bug where we show a success message even if you try to increase (or decrease) beyond the max (or min) and so no change takes effect; something to fix later.

### CODE REVIEW

File: site/src/pages/WorkspacePage/WorkspacePage.tsx [hunk 1]
```diff
@@ -65,86 +33,10 @@ export const WorkspacePage: FC = () => {
         {Boolean(checkPermissionsError) && <ErrorSummary error={checkPermissionsError} />}
       </div>
     )
-  } else if (!workspace || !permissions) {
-    return <FullScreenLoader />
-  } else if (!template) {
-    return <FullScreenLoader />
+  } else if (workspace && workspaceState.matches("ready")) {
```
Comment (BrunoQuaresma) [inline site/src/pages/WorkspacePage/WorkspacePage.tsx:36]: Is not something we could use the Choose one component?

File: site/src/xServices/workspace/workspaceXService.ts [hunk 2]
```diff
@@ -551,6 +568,31 @@ export const workspaceMachine = createMachine(
...
+      //   }
+      //   const scheduleBannerRef = spawn(workspaceScheduleBannerMachine.withContext(scheduleContext), "scheduleBannerMachine")
+      //   return { ...context, scheduleBannerRef }
+      // }),
+      // stopScheduleBannerRef: (context) => {
+      //   context.scheduleBannerRef?.stop && context.scheduleBannerRef.stop()
+      // },
+      // clearScheduleBannerRef: assign({
+      //   scheduleBannerRef: (_) => undefined
+      // }),
```
Comment (BrunoQuaresma) [inline site/src/xServices/workspace/workspaceXService.ts:588]: Should we remove these?

**Review by BrunoQuaresma — APPROVED** [review]
LGTM. Nice refactoring!

File: site/src/xServices/workspace/workspaceXService.ts (hunk 2 above)
Comment (presleyp) [reply site/src/xServices/workspace/workspaceXService.ts:588]: yes thank you! Makes me realize - I tried both ways, spawning and invoking, and I wrote that I spawned but I really invoked! Haha, you can see why, invoking requires less code.

File: site/src/pages/WorkspacePage/WorkspacePage.tsx (hunk 1 above)
Comment (presleyp) [reply site/src/pages/WorkspacePage/WorkspacePage.tsx:36]: Ooh good catch!

**Pull request merged**